blas/tpls/KokkosBlas1_nrm1_tpl_spec_decl.hpp

```diff
@@ -39,161 +39,77 @@ inline void nrm1_print_specialization() {
-        int N   = numElems;                                                    \
-        int one = 1;                                                           \
-        R()     = HostBlas<double>::asum(N, X.data(), one);                    \
-      } else {                                                                 \
-        Nrm1<ExecSpace, RV, XV, 1, false, ETI_SPEC_AVAIL>::nrm1(space, R, X);  \
-      }                                                                        \
-      Kokkos::Profiling::popRegion();                                          \
-    }                                                                          \
-  };
-
-#define KOKKOSBLAS1_SNRM1_TPL_SPEC_DECL_BLAS(LAYOUT, MEMSPACE, ETI_SPEC_AVAIL) \
-  template <class ExecSpace>                                                   \
+#define KOKKOSBLAS1_NRM1_TPL_SPEC_DECL_BLAS(SCALAR, LAYOUT, EXECSPACE,         \
+                                            MEMSPACE)                          \
+  template <>                                                                  \
   struct Nrm1<                                                                 \
-      ExecSpace,                                                               \
-      Kokkos::View<float, LAYOUT, Kokkos::HostSpace,                           \
-                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
-      Kokkos::View<const float*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>,  \
-                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
-      1, true, ETI_SPEC_AVAIL> {                                               \
-    typedef Kokkos::View<float, LAYOUT, Kokkos::HostSpace,                     \
-                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
-        RV;                                                                    \
-    typedef Kokkos::View<const float*, LAYOUT,                                 \
-                         Kokkos::Device<ExecSpace, MEMSPACE>,                  \
-                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
-        XV;                                                                    \
-    typedef typename XV::size_type size_type;                                  \
+      EXECSPACE,                                                               \
+      Kokkos::View<typename Kokkos::ArithTraits<SCALAR>::mag_type, LAYOUT,     \
+                   Kokkos::HostSpace,                                          \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,                   \
+      Kokkos::View<const SCALAR*, LAYOUT, Kokkos::Device<EXECSPACE, MEMSPACE>, \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,                   \
+      1, true,                                                                 \
+      nrm1_tpl_spec_avail<                                                     \
```

Comment:
```suggestion
      nrm1_eti_spec_avail<                                                     \
```
And since ExecSpace isn't a free template parameter, leaving this last template parameter should work too